api/v1alpha7/types.go

```diff
@@ -104,6 +104,25 @@ type SubnetFilter struct {
 	NotTagsAny      string `json:"notTagsAny,omitempty"`
 }
 
+type RouterFilter struct {
+	ID           string `json:"id,omitempty"`
+	Name         string `json:"name,omitempty"`
+	Description  string `json:"description,omitempty"`
+	AdminStateUp *bool  `json:"admin_state_up,omitempty"`
+	Distributed  *bool  `json:"distributed,omitempty"`
```

Comment:
I'm less sure about `Distributed`, but I'm inclined to say it's not relevant to what we're filtering on here, so we should omit it.

Reply:
I think this is an immutable property, but it's hard to imagine case when filtering using this makes sense.